GuideForNewDevelopers.md

```diff
@@ -0,0 +1,579 @@
+
+```sh
+cd physionet-build
+```
+Now create the virtual environment with the following command in the same terminal
+```sh
+python3 -m venv env
+```
+
+##### 2.3.6.2 Activate the Virtual Environment
+
+In the same terminal from  step [2.3.6.1](#2361-create-a-virtual-environment), enter the following command to activate the virtual environment
+
+```sh
+source env/bin/activate
+```
+
+
+#### 2.3.7 Install requirements
+
+Now that we have the Virtual Environment setup, let's install the python libraries needed for the project.
+
+In the same terminal from step [2.3.6.2](#2362-activate-the-virtual-environment), enter the following command to install the requirements
+
+```sh
+pip install -r requirements.txt
+```
+
+#### 2.3.8 Run the project
+
+We now have everything set up to run the project locally. Let's go ahead and set up the project
+
+##### 2.3.8.1 Open a terminal and navigate inside the project directory
+
+```sh
+cd physionet-build
+```
+
+
+
+##### 2.3.8.2 Activate virtual python environment.
+
+```sh
+source env/bin/activate
+```
+
+##### 2.3.8.3 Copy .env.example file to .env
+
+```sh
+cp .env.example .env
+```
+
+
+##### 2.3.8.4 Run the following commands to setup the database
+
+  - Run: `python3 manage.py resetdb` to reset the database with the latest applied migrations.
+
+  - Run: `python3 manage.py loaddemo` to load the demo fixtures set up example files.
+
+  - Run: `python3 manage.py compilestatic` to compile the css files
+
+
+#### 2.3.9 Start the server(Finally)
+
+Enter the following command to start the server
+
+```sh
+python3 manage.py runserver
+```
+
+The local development server will be available at [http://localhost:8000](http://localhost:8000).
+
+
+
+## Working on new features
+
+Now that you have completed your setup and have familiarized yourself with the codebase, here is how you can contribute and submit your changes for review.
+
+1. Create a new Branch for each feature you work on
+2. Work on your changes, add files and commit your changes
+3. On Commits, add a clear explanation of the changes. Here is an interesting read from [freeCodeCamp](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/)
+4. Once your changes are final and ready to be submitted, push the changes and open a Merge Request. Someone will review your changes ASAP.
+
+
+
+
+Here are some good resources to read about contributing to OpenSource projects
+1. [Making your first Open Source Pull Request | Github](https://www.geeksforgeeks.org/making-first-open-source-pull-request/)
+2. [A First Timers Guide to an Open Source Project](https://auth0.com/blog/a-first-timers-guide-to-an-open-source-project/)
+3. [Contributing to Open Source : Getting Started](https://www.geeksforgeeks.org/contributing-to-open-source-getting-started)
+4. [The (written) unwritten guide to pull requests
+](https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests)
+
```

Comment:
how about including this link on Python's style guide?
https://peps.python.org/pep-0008/

Reply:
Excellent idea, i will add that and submit a comment later today. 👍

Reply:
Totally agreed! I would add that comment just as you explained to me, so that people get a good sense of why would using docker be worth the "trouble"